lib/icingadb/redisconnection.hpp

```diff
@@ -231,6 +239,64 @@ namespace icinga
 		Ptr m_Parent;
 	};
 
+template<class T>
+bool RedisConnection::AwaitResultOfQuery(RedisConnection::Query query, RedisConnection::QueryPriority priority, T timeout)
+{
```

Comment:
Why not take the existing `GetResultOfQuery` function and make it return a `future`? This could then be renamed to something like `FutureForQuery` and `GetResultOfQuery` could simply be implemented as `return FutureForQuery(...).get();`.

This would not duplicate so much code and would also be way more flexible for the caller as they can just do everything you can do with a future.